### PR TITLE
Update DALI PyTorch RN50 example to the latest AMP version

### PR DESCRIPTION
**Pull request by JanuszL**
- updated DALI PyTorch RN50 example to the currently recommended way  of using AMP
- https://github.com/NVIDIA/apex/issues/511

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It updates DALI PyTorch RN50 example to the latest AMP version

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     updated DALI PyTorch RN50 example to the currently recommended way  of using AMP
 - Affected modules and functionalities:
     PyTorch RN50 example
 - Key points relevant for the review:
     NA
 - Validation and testing:
     CI L1 and L3 test
 - Documentation (including examples):
     NA

https://github.com/NVIDIA/apex/issues/511

**JIRA TASK**: *[NA]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

File: docs/examples/use_cases/pytorch/resnet50/main.py [hunk 1]
```diff
@@ -461,14 +519,14 @@ def adjust_learning_rate(optimizer, epoch, step, len_epoch):
...
 
     """Warmup"""
     if epoch < 5:
-        lr = lr * float(1 + step + epoch * len_epoch) / (5. * len_epoch)
+        lr = lr*float(1 + step + epoch*len_epoch)/(5.*len_epoch)
 
-    if(args.local_rank == 0 and step % args.print_freq == 0 and step > 1):
-        print("Epoch = {}, step = {}, lr = {}".format(epoch, step, lr))
+    # if(args.local_rank == 0):
+    #     print("epoch = {}, step = {}, lr = {}".format(epoch, step, lr))
```
Comment (klecki) [inline docs/examples/use_cases/pytorch/resnet50/main.py:529]: Why is this commented out?

Comment (JanuszL) [reply docs/examples/use_cases/pytorch/resnet50/main.py:529]: It is also in https://github.com/NVIDIA/apex/blob/master/examples/imagenet/main_amp.py. Remvoed.

File: docs/examples/use_cases/pytorch/resnet50/main.py [hunk 2]
```diff
@@ -24,59 +26,101 @@
...
+                        help='Launch test mode with preset arguments')
+    args = parser.parse_args()
+    return args
+
+best_prec1 = 0
+args = parse()
+
+print(args.test)
+# test mode, use default args for sanity test
+if args.test:
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply docs/examples/use_cases/pytorch/resnet50/main.py:91]: Added

File: docs/examples/use_cases/pytorch/resnet50/main.py [hunk 3]
```diff
@@ -24,59 +26,101 @@
...
+    args.distributed = int(os.environ['WORLD_SIZE']) > 1
+
+# make apex optional
+if args.opt_level is not None or args.distributed or args.sync_bn:
+    try:
+        from apex.parallel import DistributedDataParallel as DDP
+        from apex.fp16_utils import *
+        from apex import amp, optimizers, parallel
+    except ImportError:
+        raise ImportError("Please install apex from https://www.github.com/nvidia/apex to run this example.")
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply docs/examples/use_cases/pytorch/resnet50/main.py:116]: Moved

File: docs/examples/use_cases/pytorch/resnet50/main.py [hunk 4]
```diff
@@ -24,59 +26,101 @@
...
+    parser.add_argument('--keep-batchnorm-fp32', type=str, default=None)
+    parser.add_argument('--loss-scale', type=str, default=None)
+    parser.add_argument('--channels-last', type=bool, default=False)
+    parser.add_argument('-t', '--test', action='store_true',
+                        help='Launch test mode with preset arguments')
+    args = parser.parse_args()
+    return args
+
+best_prec1 = 0
+args = parse()
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply docs/examples/use_cases/pytorch/resnet50/main.py:87]: Fixed

File: docs/examples/use_cases/pytorch/resnet50/main.py [hunk 5]
```diff
@@ -24,59 +26,101 @@
...
+
+print(args.test)
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply docs/examples/use_cases/pytorch/resnet50/main.py:89]: Removed

File: docs/examples/use_cases/pytorch/resnet50/pytorch-resnet50.rst [hunk 6]
```diff
@@ -55,7 +55,7 @@ Usage
    positional arguments:
    DIR                         path(s) to dataset (if one path is provided, it is assumed to have subdirectories named "train" and "val"; alternatively, train and val paths can be specified directly by providing both paths as arguments)
 
-   optional arguments:
+   optional arguments (for the full list please check `APEx ImageNet example <https://github.com/NVIDIA/apex/tree/master/examples/imagenet>`_)
```
Comment (klecki) [inline docs/examples/use_cases/pytorch/resnet50/pytorch-resnet50.rst:58]: Is it spelled APEx or is it a typo?

Comment (JanuszL) [reply docs/examples/use_cases/pytorch/resnet50/pytorch-resnet50.rst]: I don't remember why I put it here like this. Changed to `Apex`.

File: docs/examples/use_cases/pytorch/resnet50/main.py [hunk 7]
```diff
@@ -497,4 +587,4 @@ def reduce_tensor(tensor):
     return rt
 
 if __name__ == '__main__':
-    main()
+    main()
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply docs/examples/use_cases/pytorch/resnet50/main.py]: Done

**Pull request merged**